### PR TITLE
Optional self-supervised pre-training

### PR DESCRIPTION
**Pull request by athewsey**
Refactor LayoutLM model training & inference code to support MLM
pre-training before NER fine-tuning, and present steps in notebook.
Upgrade HF container versions 4.6->4.11.

**Issue #, if available:** #3

**Description of changes:**

- Refactor LayoutLM model training and inference code to support masked language modelling pre-training before the NER fine-tuning
- Add optional steps & guidance in notebook 2 to run the pre-training and use the result as the basis for NER fine-tuning
- Upgrade Hugging Face container versions 4.6->4.11 (some API changes involved, may as well align to the more current version)

**Testing done:**

Tested a variety of pre-training configurations & impact on down-stream entity recognition accuracy, in a pre-existing environment.

<br/>

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by athewsey — COMMENTED** [review]
All looks good, seems to work and not to introduce anything inconsistent with the original blog
